cirq/circuits/circuit_test.py

```diff
@@ -27,6 +27,40 @@
 import cirq.google as cg
 
 
+def raise_value_error_if_wrong_type(obj, classinfo):
+    if not isinstance(obj, classinfo):
+        raise ValueError('not isinstance({!r}, {!r})'.format(obj, classinfo))
+
+
+class _MomentAndOpTypeValidatingDevice(
+        cirq.devices.unconstrained_device._UnconstrainedDeviceType):
```

Comment:
Instead of inheriting from an unrelated and hidden class, just inherit directly from Device.